File: FilFeed/src/androidTest/java/com/phicdy/mycuration/util/test/TextUtilTest.java
package com.phicdy.mycuration.util.test;

import android.test.AndroidTestCase;

import com.phicdy.mycuration.util.TextUtil;

public class TextUtilTest extends AndroidTestCase {

	public TextUtilTest() {
		super();
	}

	protected void setUp() throws Exception {
		super.setUp();
	}

	protected void tearDown() throws Exception {
		super.tearDown();
	}

	public void testRemoveLineFeed() {
		assertEquals("aaa", TextUtil.removeLineFeed("aaa\r"));
		assertEquals("aaa", TextUtil.removeLineFeed("aaa\n"));
		assertEquals("aaa", TextUtil.removeLineFeed("aaa\t"));
		assertEquals("aaa", TextUtil.removeLineFeed("aaa\r\n"));

		assertEquals("aaa", TextUtil.removeLineFeed("a\ra\na"));
		assertEquals("aaa", TextUtil.removeLineFeed("a\ra\ta"));
		assertEquals("aaa", TextUtil.removeLineFeed("a\ra\r\na"));
		assertEquals("aaa", TextUtil.removeLineFeed("a\na\ta"));
		assertEquals("aaa", TextUtil.removeLineFeed("a\na\r\na"));
		assertEquals("aaa", TextUtil.removeLineFeed("a\ta\r\na"));
	}
}

File: domain/src/test/java/com/phicdy/mycuration/domain/util/TextUtilTest.kt
package com.phicdy.mycuration.domain.util

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class TextUtilTest {

    @Test
    fun testRemoveLineFeed() {
        assertThat(TextUtil.removeLineFeed("aaa\r")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("aaa\n")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("aaa\t")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("aaa\r\n")).isEqualTo("aaa")

        assertThat(TextUtil.removeLineFeed("a\ra\na")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("a\ra\ta")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("a\ra\r\na")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("a\na\ta")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("a\na\r\na")).isEqualTo("aaa")
        assertThat(TextUtil.removeLineFeed("a\ta\r\na")).isEqualTo("aaa")
    }
}
